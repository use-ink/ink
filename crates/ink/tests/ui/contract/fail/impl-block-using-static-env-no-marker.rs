#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self::constructor_impl()
        }
    }

    impl Contract {
        #[ink(message)]
        pub fn message(&self) {}
    }

    impl Contract {
        fn constructor_impl() -> Self {
            let _ = Self::env().caller();
            Self {}
        }
    }
}

fn main() {}
