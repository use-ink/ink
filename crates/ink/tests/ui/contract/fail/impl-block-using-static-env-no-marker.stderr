error[E0599]: no function or associated item named `env` found for struct `Contract` in the current scope
  --> tests/ui/contract/fail/impl-block-using-static-env-no-marker.rs:20:27
   |
4  |     pub struct Contract {}
   |     ------------------- function or associated item `env` not found for this struct
...
20 |             let _ = Self::env().caller();
   |                           ^^^ function or associated item not found in `Contract`
   |
note: if you're trying to build a new `Contract` consider using one of the following associated functions:
      contract::_::<impl Contract>::constructor
      Contract::constructor_impl
  --> tests/ui/contract/fail/impl-block-using-static-env-no-marker.rs:8:9
   |
8  |         pub fn constructor() -> Self {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
19 |         fn constructor_impl() -> Self {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: items from traits can only be used if the trait is in scope
help: trait `StaticEnv` which provides `env` is implemented but not in scope; perhaps you want to import it
   |
1  + use ink::codegen::StaticEnv;
   |
